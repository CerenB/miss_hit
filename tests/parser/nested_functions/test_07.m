% From https://www.mathworks.com/help/matlab/matlab_prog/nested-functions.html

function p = test_07(a,b,c)
    p = @parabola;

    function y = parabola(x)
        y = a*x.^2 + b*x + c;
    end

end
