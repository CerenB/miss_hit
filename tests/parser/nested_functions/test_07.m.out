----------------------------------------------------------------------
--  Parse tree for test_07.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_07>
      Inputs: Identifier <a>
      Inputs: Identifier <b>
      Inputs: Identifier <c>
      Outputs: Identifier <p>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Identifier <p>
        RHS: Function_Pointer
          Name: Identifier <parabola>
    Nested: Function_Definition
      Signature: Function_Signature
        Name: Identifier <parabola>
        Inputs: Identifier <x>
        Outputs: Identifier <y>
      Body: Sequence_Of_Statements
        Statements: Simple_Assignment_Statement
          LHS: Identifier <y>
          RHS: Binary_Operation +
            LHS: Binary_Operation +
              LHS: Binary_Operation *
                LHS: Identifier <a>
                RHS: Binary_Operation .^
                  LHS: Identifier <x>
                  RHS: Number_Literal <2>
              RHS: Binary_Operation *
                LHS: Identifier <b>
                RHS: Identifier <x>
            RHS: Identifier <c>
----------------------------------------------------------------------
In test_07.m, line 3
| function p = test_07(a,b,c)
|            ^ info: this token is not linked to the ast
In test_07.m, line 3
| function p = test_07(a,b,c)
|                     ^ info: this token is not linked to the ast
In test_07.m, line 3
| function p = test_07(a,b,c)
|                       ^ info: this token is not linked to the ast
In test_07.m, line 3
| function p = test_07(a,b,c)
|                         ^ info: this token is not linked to the ast
In test_07.m, line 3
| function p = test_07(a,b,c)
|                           ^ info: this token is not linked to the ast
In test_07.m, line 4
|     p = @parabola;
|                  ^ info: this token is not linked to the ast
In test_07.m, line 6
|     function y = parabola(x)
|                ^ info: this token is not linked to the ast
In test_07.m, line 6
|     function y = parabola(x)
|                          ^ info: this token is not linked to the ast
In test_07.m, line 6
|     function y = parabola(x)
|                            ^ info: this token is not linked to the ast
In test_07.m, line 7
|         y = a*x.^2 + b*x + c;
|                             ^ info: this token is not linked to the ast
In test_07.m, line 8
|     end
|     ^^^ info: this token is not linked to the ast
In test_07.m, line 10
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
